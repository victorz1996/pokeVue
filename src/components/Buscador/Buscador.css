.sizeInput {
    width: 700px;
}