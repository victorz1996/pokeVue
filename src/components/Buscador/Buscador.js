/**
 * @namespace Buscador
 * @description Componente para realizar la busqueda de pokemones
 * @author       Victor Zabala <victorma164@gmail.com>
 */
export default {
    name: 'Buscador',
    data() {
        return {
            buscarInp: '',
        }
    },
    methods: {
        /**
        * @memberOf Buscador
        * @function filtrar funcion de filtro
        * @description Metodo que emite lo que el usuario escribe en el input de busqueda
        * para disparar la funcion de busqueda.
        */
        filtrar() {
            this.$emit('filtro', this.buscarInp)
        }
    },
}