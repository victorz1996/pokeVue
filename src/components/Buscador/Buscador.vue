<template>
  <div class="input-group input-group-lg sizeInput">
    <span class="input-group-text"><font-awesome-icon icon="search" /></span>
    <input
      type="text"
      class="form-control"
      placeholder="Search"
      @input="filtrar"
      v-model="buscarInp"
      aria-label="Search"
    />
  </div>
</template>
<script src="./Buscador.js"></script>
<style lang="css" src="./Buscador.css"></style>