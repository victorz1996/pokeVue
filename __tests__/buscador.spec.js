import { shallowMount } from '@vue/test-utils'
import Buscador from "../src/components/Buscador/Buscador.vue"

let cmp

    // eslint-disable-next-line no-undef
    beforeEach(() => {
      cmp = shallowMount(Buscador)
    })
    // eslint-disable-next-line no-undef
    describe('Componente buscador', () => {
      // eslint-disable-next-line no-undef
      test('Instancia de Vue.js',() => {
        // eslint-disable-next-line no-undef
        expect(cmp.isVueInstance).toBeTruthy()
      })

      // eslint-disable-next-line no-undef
      it('HTMl renderizado structure esperada',()=>{
        // eslint-disable-next-line no-undef
        expect(cmp.element).toMatchSnapshot()
    })
})