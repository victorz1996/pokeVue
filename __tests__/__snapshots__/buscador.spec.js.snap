// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente buscador HTMl renderizado structure esperada 1`] = `
<div
  class="input-group input-group-lg sizeInput"
>
  <span
    class="input-group-text"
  >
    <font-awesome-icon
      icon="search"
    />
  </span>
   
  <input
    aria-label="Search"
    class="form-control"
    placeholder="Search"
    type="text"
  />
</div>
`;
